app/models/concerns/starter_code_importable.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+module StarterCodeImportable
+  extend ActiveSupport::Concern
+
+  def starter_code?
+    starter_code_repo_id.present?
+  end
+
+  def starter_code_repository
+    return unless starter_code?
+    @starter_code_repository ||= GitHubRepository.new(creator.github_client, starter_code_repo_id)
+  end
+
+  def use_template_repos?
+    starter_code? && template_repos_enabled?
+  end
+
+  def use_importer?
+    starter_code? && !template_repos_enabled?
+  end
+
+  def starter_code_repository_is_a_template_repository
+    return unless use_template_repos?
+
+    options = { accept: "application/vnd.github.baptiste-preview" }
+    endpoint_url = "https://api.github.com/repositories/#{starter_code_repo_id}"
+    starter_code_github_repository = creator.github_client.get(endpoint_url, options)
+
+    return if starter_code_github_repository.is_template
+    errors.add(
+      :starter_code_repository,
+      "is not a template repository. Make it a template repository to use template cloning."
```

Comment:
I'd like to point to documentation here, but this can be done in the future PR.

Reply:
I was planning to do that, but wouldn't that require adding HTML markup to the error message to link to the docs? It's easy/doable but it might get ugly for tests

Reply:
I'd like to err on the side of UX over having all our tests be 100% pretty